src/server/v2.0/handler/scanexport.go

```diff
@@ -0,0 +1,202 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/controller/scandataexport"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/pkg/scan/export"
+	v1 "github.com/goharbor/harbor/src/pkg/scan/rest/v1"
+	"github.com/goharbor/harbor/src/pkg/systemartifact"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/scan_data_export"
+)
+
+func newScanDataExportApi() *scanDataExportAPI {
+	return &scanDataExportAPI{
+		scanDataExportCtl: scandataexport.Ctl,
+		sysArtifactMgr:    systemartifact.Mgr,
+	}
+}
+
+type scanDataExportAPI struct {
+	BaseAPI
+	scanDataExportCtl scandataexport.Controller
+	sysArtifactMgr    systemartifact.Manager
+}
+
+func (se *scanDataExportAPI) Prepare(ctx context.Context, operation string, params interface{}) middleware.Responder {
+	return nil
+}
+
+func (se *scanDataExportAPI) ExportScanData(ctx context.Context, params operation.ExportScanDataParams) middleware.Responder {
+	if err := se.RequireAuthenticated(ctx); err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	// check if the MIME type for the export is the Generic vulnerability data
+	if params.XScanDataType != v1.MimeTypeGenericVulnerabilityReport {
+		error := &models.Error{Message: fmt.Sprintf("Unsupported MIME type : %s", params.XScanDataType)}
+		errors := &models.Errors{Errors: []*models.Error{error}}
+		return operation.NewExportScanDataBadRequest().WithPayload(errors)
+	}
+
+	// loop through the list of projects and validate that scan privilege and create privilege
+	// is available for all projects
+	// TODO : Should we just ignore projects that do not have the required level of access?
+
+	projects := params.Criteria.Projects
+	for _, project := range projects {
+		if err := se.RequireProjectAccess(ctx, project, rbac.ActionCreate, rbac.ResourceScan); err != nil {
+			return se.SendError(ctx, err)
+		}
+	}
+
+	scanDataExportJob := new(models.ScanDataExportJob)
+
+	// vendor id associated with the execution can be set to -1
+	// is not associated with the state of any other scan report entity
+	userContext := context.WithValue(ctx, "vendorId", -1)
```

Comment:
hardcode it in the scanDataExportCtl?

Reply:
If the vendorId is -1, the execution will only be retained as 50 for all users, so we could use userId as the vendorId to let every user can keep self 50 records. Ref: https://github.com/goharbor/harbor/blob/dc1a03fde25cc05f82705e13106ba9cfae8fc918/src/pkg/task/execution.go#L146

Reply:
@chlins  Based on my earlier discussions, we had decided to keep the vendor id as -1. Is there a library method to get the user id for the current requesting user irrespective of the auth mechanism?

Reply:
@wy65701436  I will hold on to this comment. If we want to pass the user id as the vendor id then we will have to do it within this location itself. I wanted to understand the possible implications of using  -1 other than the fact that executions would be retained across all users as opposed to specific user ids.

Reply:
I think why @chlins purpose to use the userid as the vendor id is to avoid the execute sweeper to sweep the valid exeuctions.

Some backgound: 
To avoid having too many execution records in the database, Harbor introduced an execution sweeper to clean up what we consider to be useless execution records. When a user triggers a new exeuction, the sweeper starts sweeping the execution records. The key point here is that the sweeper uses the vendor ID and vendor type as query criteria to query the exeuction records and keep the latest 50. This means that if we use -1 as the vendor ID for CSV export execution, the sweeper will keep the latest 50 reports for all users. However, if we have hunderds of users and at least one CVE report per person, then 50 entries are not enough here.